qemu/qmp.go

```diff
@@ -1255,6 +1255,46 @@ func (q *QMP) ExecHotplugMemory(ctx context.Context, qomtype, id, mempath string
 	return err
 }
 
+// ExecuteNVDIMMDeviceAdd adds the guest portion of a block device to
+// a QEMU instance using a NVDIMM driver with the device_add command.
+// id is the id of the device to add.  It must be valid QMP identifiers.
+// mempath is the path of the device to add, e.g., /dev/rdb0.  size is
+// the data size of the device.
+func (q *QMP) ExecuteNVDIMMDeviceAdd(ctx context.Context, id, mempath string, size int64) error {
+	args := map[string]interface{}{
+		"qom-type": "memory-backend-file",
+		"id":       "nvdimmbackmem" + id,
+		"props": map[string]interface{}{
+			"mem-path": mempath,
+			"size":     size,
+			"share":    true,
+		},
+	}
+	err := q.executeCommand(ctx, "object-add", args, nil)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
```

Comment:
I'm not sure it's really worth having a defer here, unless you're planning to add more commands to this function in the future.  It just complicates things (see the issue regarding the error below) and slows things down.